packages/block-library/src/navigation-submenu/edit.js

```diff
@@ -317,23 +317,19 @@ export default function NavigationSubmenuEdit( {
 	} = useSelect(
 		( select ) => {
 			const {
-				getClientIdsOfDescendants,
 				hasSelectedInnerBlock,
 				getSelectedBlockClientId,
 				getBlockParentsByBlockName,
 				getBlock,
+				getBlockCount,
+				getBlockOrder,
 			} = select( blockEditorStore );
 
 			let _onlyDescendantIsEmptyLink;
 
 			const selectedBlockId = getSelectedBlockClientId();
 
-			const descendants = getClientIdsOfDescendants( [ clientId ] )
-				.length;
-
-			const selectedBlockDescendants = getClientIdsOfDescendants( [
-				selectedBlockId,
-			] );
+			const selectedBlockDescendants = getBlockOrder( selectedBlockId );
```

Comment:
I'm not sure if this is the best replacement, but it worked in my tests.

Reply:
I guess the difference is that `getClientIdsOfDescendants` creates a deep list of clientIds (all client ids in the hierarchy below the block), and `getBlockOrder` is shallow?

It doesn't seem to matter, because the code only checks for one child.

It might be good to rename some of the variables 'child' as 'descendants' always implies a deep search to me.

The other thing I'm wondering about is why it uses `selectedBlockClientId` rather than the block's own clientId. It seems like this is used for the 'remove submenu' toolbar button that will only show if the current block is the selected one.

edit: actually, it does show when more than one submenu is selected, but seems unusable. Maybe it needs to be in a specific toolbar group.